docs/development/verilog_modules.md

```diff
@@ -0,0 +1,119 @@
+for details):
+
+- `input clk_i`: LPC clock.
+- `input nrst_i`: LPC reset (active low).
+- `input lframe_i`: LPC frame input (active low).
+- `inout lad_bus`: LPC data bus, slow pull-up on host side.
+- `inout serirq`: LPC SERIRQ signal, slow pull-up on host side.
+
+Ports for signals to/from data provider:
+
+- `output lpc_addr_o`: 16-bit address of TPM register.
+- `inout lpc_data_io`: data received from (write) or to be sent (read) to host.
+  See descriptions of next 4 signals for requirements related to driving this
+  signal.
+- `output lpc_data_wr`: signal to data provider that `lpc_data_io` is driven by
+  LPC module. Data provider should stop driving `lpc_data_io` immediately upon
+  receiving this signal. To avoid hazards, data provider should sample data on
+  rising edge of `clk_i` or wait at least half of `clk_i` cycle (15 ns for 33
+  MHz) before sampling.
+- `input lpc_wr_done`: signal from data provider that `lpc_data_io` has been
+  read. This signal should be driven until LPC module stops driving
+  `lpc_data_wr`, after which this signal should be stopped being driven no later
+  than 1 `clk_i` cycle (30ns). LPC module changes `lpc_data_wr` on falling
+  `clk_i` edge so it is suggested that data provider module samples that signal
+  on rising edge.
+- `output lpc_data_req`: signal to data provider that data is requested (rising
+  edge of this signal) or has been read (falling edge) from `lpc_data_io`.
+  `lpc_data_req` is changed on falling edge of `clk_i`.
+- `input lpc_data_rd`: signal from data provider that `lpc_data_io` has valid
+  data for reading. This signal should be driven in response to `lpc_data_req`.
+  It should never be driven if `lpc_data_wr` is also active. `lpc_data_rd` is
+  sampled by LPC module on falling `clk_i` edge. Data provider module may either
+  start driving both `lpc_data_rd` and `lpc_data_io` signals on rising edge, or
+  drive `lpc_data_io` for at least half a cycle (15 ns) before asserting
+  `lpc_data_rd`. This signal should be driven until LPC module stops driving
+  `lpc_data_req`, after which this signal should be stopped being driven no
+  later than 1 `clk_i` cycle (30ns). LPC module changes `lpc_data_req` on
+  falling `clk_i` edge so it is suggested that data provider module samples that
+  signal on rising edge.
+- `input irq_num`: IRQ number, for TPM this is `TPM_INT_VECTOR_x.sirqVec`. No
+  parsing is done by LPC module, meaning that both IRQ0 (in some parts of TPM
+  specification this means "IRQ disabled") and IRQ2 (SMI) are valid. IRQ number
+  is sampled on `clk_i` falling edge on SERIRQ start frame, turn-around phase.
+  This is done to avoid sending one interrupt with two different IRQ numbers in
+  one cycle.
+- `input interrupt`: whether interrupt should be signaled to host, active high.
+  It is checked at the beginning of IRQ frame of IRQ latched from `irq_num`,
+  both in quiet and continuous mode. In addition to that, in quiet mode this
```

Comment:
Again SERIRQ quiet and continuous mode should be irrelevant to data provider, yet we introduce those without explanation or links to SERIRQ specs

Reply:
Link added.